src/backend/distributed/utils/shard_utils.c

```diff
@@ -55,3 +56,21 @@ GetLongestShardName(Oid citusTableOid, char *finalRelationName)
 
 	return longestShardName;
 }
+
+
+/*
+ * GetLongestShardName is a utility function that creates a hypothetical shard
+ * name for a table that is not distributed yet.
+ *
+ * Note that this operation will increment the shard id seqeunce without
+ * actually using the shard id in an actual shard.
+ */
+char *
+GetLongestHypotheticalShardName(char *relationName)
```

Comment:
I implemented what you suggested, with a small change.

> shardCount of the parentTable
instead of this, I used the current `citus.shard_count` config value as accessing the shardCount of the parent table will not be as clean and readable IMO.

Also `LoadShardIntervalWithLongestShardName` returns the largest shardId of the parent table. Even if we add 1 to this value, we will hit the shardId of the next table that is distributed after this parent table.